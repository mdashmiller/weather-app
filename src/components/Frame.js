import styled from "styled-components"

const Frame = styled.div`
	width: 300px;
	margin: 40px auto 30px;
	text-align: center;
	font-family: 'Varela Round', Arial, sans-serif;
`

export default Frame
