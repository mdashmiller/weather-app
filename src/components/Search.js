import styled, { css } from "styled-components"

const Search = styled.input`
	background: #eee;
	border: 1px solid #c3c3c3;
	border-radius: 10px;
	margin: 200px auto 20px;
	width: 200px;
	height: 34px;
	padding-left: 8px;
	font-family: 'Font Awesome 5 Free', 'Varela Round', Arial, sans-serif;
	font-weight: 900;
	font-size: 18px;

	&::placeholder {
		color: #6a6a6a;
	}

	${props => props.short && css`
	    width: 114px;
	    padding-left: 5px;
	    margin: 0;
	  `}

	${props => props.landingPage && css`
		display: block;
	    width: 172px;
	   	margin: 26px auto;
		text-align: center;
		padding: 0;
		cursor: pointer;  
	  `}

	${props => props.userLocation && css`
		border: 3px solid #f46403;
			
			&:hover {
				background: #f46403;
				
				&::placeholder {
					color: #eee;
				}
			}	
	  `}

	${props => props.userZip && css`
		border: 3px solid #6a6a6a;

			&:hover {
				background: #6a6a6a;
				
				&::placeholder {
					color: #eee;
				}
			} 
	  `}
`

export default Search
