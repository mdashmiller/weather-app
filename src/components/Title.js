import React from 'react'
import { Link } from 'react-router-dom'

const Title = () =>
	<Link to='/'>
		<h1 id="landing-title" className="title-gold">
			Weather
			<span className="title-grey">
				Now
			</span>
		</h1>
	</Link>

export default Title
