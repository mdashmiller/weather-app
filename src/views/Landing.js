import React from 'react'
import Frame from '../components/Frame'
import Search from '../components/Search'
import { Link } from 'react-router-dom'


const Landing = () => 
	<Frame>
		<h1 id="landing-title" className="day">
			Weather
			<span className="day-2">
				Now
			</span>
		</h1>
		<Link to='/weather-by-geoloc'>			    	
			<Search
				type="text"
				placeholder="&#xf3c5;  Use My Location"
				landingPage
				userLocation
			/>
		</Link>
		<Link to='/lookup'>
			<Search
				placeholder="&#xf002;  Lookup by Zip"
				landingPage
				userZip
			/>		 		    
		</Link>    
	</Frame>

	

export default Landing
