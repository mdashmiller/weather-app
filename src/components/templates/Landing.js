import React from 'react'
import { Link } from 'react-router-dom'
import Frame from '../ui/Frame'
import Search from '../ui/Search'
import Title from '../ui/Title'

const Landing = () => 
	<Frame>
		<Title />
		<Link to='/lookup-by-geoloc'>			    	
			<Search
				type="text"
				placeholder="&#xf3c5;  Use My Location"
				landingPage
				userLocation
			/>
		</Link>
		<Link to='/lookup-by-zip'>
			<Search
				placeholder="&#xf002;  Lookup by Zip"
				landingPage
				userZip
			/>		 		    
		</Link>    
	</Frame>

export default Landing
