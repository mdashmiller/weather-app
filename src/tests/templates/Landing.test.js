import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter, Link } from 'react-router-dom'

import Landing from '../../components/templates/Landing'
import Frame from '../../components/ui/Frame'
import Search from '../../components/ui/Search'
import Title from '../../components/ui/Title'

Enzyme.configure({ adapter: new Adapter() })

describe('<Landing /> rendering', () => {

	it('renders without crashing', () => {
	    const div = document.createElement('div')
	    ReactDOM.render(
	        <BrowserRouter>
	  		    <Landing />
	  	   </BrowserRouter>,
	  	div)
	    ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<BrowserRouter>
				<Landing />
			</BrowserRouter>
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 <Frame> component', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Frame).length).toBe(1)
	})

	it('renders 1 <Title> component', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Title).length).toBe(1)
	})

	it('renders 2 <Link> elements', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Link).length).toBe(2)
	})

	it('renders 2 <Search> components', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Search).length).toBe(2)
	})

})

describe('<Landing> Link functionality', () => {

	it('contains a link to get weather by geolocation', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Link).at(0).prop('to')).toBe('/lookup-by-geoloc')
	})

	it('contains a link to get weather by zipcode', () => {
		const wrapper = shallow(<Landing />)
		expect(wrapper.find(Link).at(1).prop('to')).toBe('/lookup-by-zip')
	})

})
