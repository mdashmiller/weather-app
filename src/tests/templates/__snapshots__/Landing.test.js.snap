// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> rendering has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa cOdhsb"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="title-gold"
      id="landing-title"
    >
      Weather
      <span
        className="title-grey"
      >
        Now
      </span>
    </h1>
  </a>
  <a
    href="/lookup-by-geoloc"
    onClick={[Function]}
  >
    <input
      className="sc-bwzfXH iOThS"
      placeholder="  Use My Location"
      type="text"
    />
  </a>
  <a
    href="/lookup-by-zip"
    onClick={[Function]}
  >
    <input
      className="sc-bwzfXH jqlRDI"
      placeholder="  Lookup by Zip"
    />
  </a>
</div>
`;
